File: 0206-reverse-linked-list/0206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head:
            prev=None
            curr=head
            curr_next=curr.next
            while curr_next is not None:
                curr.next,prev,curr,curr_next=prev,curr,curr_next,curr_next.next
            curr.next=prev
            return curr
        else:
            return head

            

File: 0206-reverse-linked-list/0206-reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {

        if (head == null)
        {
            return head;
        }
        else
        {
        ListNode temp = head;
        ListNode prev = null;
        ListNode dummy = null;
        while(temp != null)
        {
            dummy = temp.next;
            temp.next = prev;
            prev = temp;
            temp = dummy;        
        }
        return prev;
        }   
    }
}